src/pages/docs/creators/ray/cluster-yaml.md

```diff
@@ -104,4 +104,88 @@ By default, Ray on Golem uses prepackaged VM images including [relatively fresh]
 Supported tags are available on [Golem registry](https://registry.golem.network/explore/golem/ray-on-golem).
 
-Please [let us know on `#Ray on Golem` discord channel)](https://chat.golem.network/) if you need an image with any specific content. We will be happy to help you.
+Please [let us know on the `#Ray on Golem` discord channel)](https://chat.golem.network/) if you need an image with any specific content. We will be happy to help you.
+
+## Budget management properties
+
+### Spending hard limit
+
+Within `provider.parameters` section there is the `budget` property.
+It defines the maximum amount of GLMs paid for the whole cluster operations - from `ray up` until `ray down`.
+
+At the moment, when the spending reaches the limit, Ray on Golem will stop spending, effectively terminating the cluster nodes.
+
+```yaml
+provider:
+  parameters:
+
+    # Maximum amount of GLMs that's going to be spent for the whole cluster
+    budget: 2
+```
+
+### Avoiding too-expensive providers
+
+You can use `provider.parameters.node_config.cost_management` section to define the limits on providers' prices.
+Ray on Golem won't work with providers who exceed any of the following price settings.
+
+#### Maximum provider prices
+
+Golem providers charge in three ways. They charge:
+- the initial price at the start of image deployment,
+- CPU usage price for the total time (in seconds) their CPUs spent computing,
+- duration price for the total time (in seconds) they spent up and running,
+
+So for example, if you rented a 3 CPU node for 15 minutes and utilized all three cores for 10 minutes you will be charged `initial_price + duration_price * 15*60 + cpu_usage_price * 3 * 10*60`.
+
+The following properties allow you to reject providers with any of the prices exceeding your limits.
+- `max_initial_price` (recommended value `0.5`)
+- `max_cpu_sec_price` (recommended value `0.0005`)
+- `max_duration_sec_price` (recommended value `0.0005`)
+
+```yaml
+provider:
+  parameters:
+    node_config:
+      cost_management:
+
+        # Amount of GLMs for worker initiation 
+        # causing Golem provider offer to be rejected
+        max_initial_price: 0.5
+
+        # Amount of GLMs for CPU utilisation per second 
+        # causing Golem provider offer to be rejected
+        max_cpu_sec_price: 0.0005
+
+        # Amount of GLMs for each second that worker runs 
+        # causing Golem provider offer to be rejected
+        max_duration_sec_price: 0.0005
+```
+
+#### Maximum average usage cost
+
+To combine all three prices into one value, have a look at the properties:
+- `average_cpu_load` (recommended value `0.8`)
+- `average_duration_minutes` (recommended value `20`)
+- `max_average_usage_cost` (recommended value `1.5`)
+
+Using those if you plan to use your cluster for 20 minutes, and keep it busy for 80% of the time, you will reject providers that would cost you more than `1.5` GLMs in total.
```

Comment:
This section also sort all offers which were not rejected and enables ray-on-golem to pick the cheapest (according to `average_cpu_load` and `average_duration_minutes`) providers first